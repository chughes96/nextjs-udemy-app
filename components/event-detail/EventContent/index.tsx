import { ReactNode } from 'react';
import styles from './event-content.module.css';

type Props = {
  children: ReactNode;
};

export const EventContent = ({ children }: Props) => {
  return <section className={styles.content}>{children}</section>;
};
