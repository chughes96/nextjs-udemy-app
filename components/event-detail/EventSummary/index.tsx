import styles from './EventSummary.module.css';

type Props = {
  title: string;
};

export const EventSummary = ({ title }: Props) => {
  return (
    <section className={styles.summary}>
      <h1>{title}</h1>
    </section>
  );
};
