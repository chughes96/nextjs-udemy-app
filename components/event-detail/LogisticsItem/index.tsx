import { ReactNode } from 'react';
import styles from './LogisticsItem.module.css';

type Props = {
  icon: React.FC;
  children: ReactNode;
};

export const LogisticsItem = ({ icon: Icon, children }: Props) => {
  return (
    <li className={styles.item}>
      <span className={styles.icon} data-testid="logistics-item-icon">
        <Icon />
      </span>
      <span className={styles.content}>{children}</span>
    </li>
  );
};
