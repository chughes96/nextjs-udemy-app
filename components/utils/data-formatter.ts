export const formatDate = (date: string) =>
  new Date(date).toLocaleDateString('en-GB', {
    day: 'numeric',
    month: 'long',
    year: 'numeric',
  });

export const formatAddress = (address: string) => address.replaceAll(', ', '\n');
