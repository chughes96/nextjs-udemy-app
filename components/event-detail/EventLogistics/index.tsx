import { EventDetails } from '@/types';
import { AddressIcon, DateIcon } from '@/components/icons';
import { LogisticsItem } from '../LogisticsItem';
import { formatAddress, formatDate } from '@/components/utils/data-formatter';
import styles from './event-logistics.module.css';

type Props = {
  eventDetails: EventDetails;
};

export const EventLogistics = ({ eventDetails: { date, location, image } }: Props) => {
  const formattedDate = formatDate(date);

  const formattedAddress = formatAddress(location);

  return (
    <section className={styles.logistics}>
      <div className={styles.image}>
        <img src={`/${image}`} alt={image} />
      </div>
      <ul className={styles.list}>
        <LogisticsItem icon={DateIcon}>
          <time>{formattedDate}</time>
        </LogisticsItem>
        <LogisticsItem icon={AddressIcon}>
          <address>{formattedAddress}</address>
        </LogisticsItem>
      </ul>
    </section>
  );
};
