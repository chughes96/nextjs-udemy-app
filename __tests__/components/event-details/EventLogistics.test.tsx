import { render, screen } from '@testing-library/react';
import { createEvent } from '@/__tests__/utils/test-data-builder';
import { getAllEvents } from '@/dummy-data';
import { EventLogistics } from '@/components/event-detail/EventLogistics';
import { formatAddress, formatDate } from '@/components/utils/data-formatter';

describe('<EventLogistics />', () => {
  let testEventDetails = createEvent();

  const renderComponent = (eventDetails = testEventDetails) => render(<EventLogistics eventDetails={eventDetails} />);

  it('renders correctly and matches the snapshot', () => {
    // Cannot use test data builder reliably for snapshots because of randomised data
    const eventDetails = getAllEvents()[0];

    const component = renderComponent(eventDetails);

    expect(component).toMatchSnapshot();
  });

  it('renders the event image', () => {
    renderComponent();

    const image = screen.getByAltText(testEventDetails.title);

    expect(image).toBeInTheDocument();
  });

  it('formats and then renders the event date', () => {
    renderComponent();

    const formattedDate = formatDate(testEventDetails.date);

    const date = screen.getByText(formattedDate);

    expect(date).toBeInTheDocument();
  });

  it('formats and then renders the event location', () => {
    renderComponent();

    const formattedLocation = formatAddress(testEventDetails.location);

    const location = screen.getByText(formattedLocation);

    expect(location).toBeInTheDocument();
  });
});
