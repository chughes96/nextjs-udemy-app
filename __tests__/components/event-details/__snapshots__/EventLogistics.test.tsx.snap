// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventLogistics /> renders correctly and matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="logistics"
      >
        <div
          class="image"
        >
          <img
            alt="images/coding-event.jpg"
            src="/images/coding-event.jpg"
          />
        </div>
        <ul
          class="list"
        >
          <li
            class="item"
          >
            <span
              class="icon"
              data-testid="logistics-item-icon"
            >
              <svg
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
            <span
              class="content"
            >
              <time>
                12 May 2021
              </time>
            </span>
          </li>
          <li
            class="item"
          >
            <span
              class="icon"
              data-testid="logistics-item-icon"
            >
              <svg
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
            <span
              class="content"
            >
              <address>
                Somestreet 25
12345 San Somewhereo
              </address>
            </span>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="logistics"
    >
      <div
        class="image"
      >
        <img
          alt="images/coding-event.jpg"
          src="/images/coding-event.jpg"
        />
      </div>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <span
            class="icon"
            data-testid="logistics-item-icon"
          >
            <svg
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
          <span
            class="content"
          >
            <time>
              12 May 2021
            </time>
          </span>
        </li>
        <li
          class="item"
        >
          <span
            class="icon"
            data-testid="logistics-item-icon"
          >
            <svg
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
          <span
            class="content"
          >
            <address>
              Somestreet 25
12345 San Somewhereo
            </address>
          </span>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
